src/django_components/extension.py

```diff
@@ -652,6 +661,56 @@ def get_extension_command(self, name: str, command_name: str) -> Type[ComponentC
                 return command
         raise ValueError(f"Command {command_name} not found in extension {name}")
 
+    def add_extension_urls(self, name: str, urls: List[URLRoute]) -> None:
```

Comment:
The basic idea is that when a component class is defined, this extension picks that up and registers a unique endpoint for this component.

But a component may be defined before or after Django apps were initialized.

- If a component was defined BEFORE the Django calls `AppsConfig.ready()`, then everything is all right and the component's URL is configured properly.

- But if the component is defined AFTER `AppsConfig.ready()`, then, even if we manually add the URL to `urlpatterns`, we still need to trigger `URLResolver._populate()`  in order to force Django to update its lookups, so things like `reverse()` work

And to make working with that easier, I defined the `Extensions.add_extension_urls()` and `Extensions.remove_extension_urls()`.